public/script.js

```diff
@@ -1,79 +1,100 @@
 
 /*
  * Emoji cheat sheet
  */
+$(document).ready(function() {
+  try {
+    if(document.flashtest && document.flashtest.PercentLoaded()>=0){
+      // Flash was able to load the test swf and is working
+      initZeroClipboard();
+    } else {
+      initJsClipboard();
+    }
+  } catch (e) {
+    initJsClipboard();
+  }
 
-$(function() {
-  if (FlashDetect.installed) {
-    var reglue;
-    var clip_flash_block_detect = new ZeroClipboard.Client();
-    clip_flash_block_detect.glue(document.getElementById('flash-test'));
-
-    var clip = new ZeroClipboard.Client();
-    clip.reposition(document.getElementById('flash-test'));
-    clip.receiveEvent('mouseover', null);
-
-    $.jnotify.setup({ 'delay': 1000, 'fadeSpeed': 500 });
+  function initZeroClipboard(){
+    ZeroClipboard.config({ 
+      forceHandCursor: true,
+      hoverClass: "hover"
+    });
+    var clipboardclient = new ZeroClipboard();
+
+    clipboardclient.on('ready', function( readyEvent ) {
+      $('ul').on('mouseover', 'div', function() {
+        try {
+          clipboardclient.clip(this);
+        } catch(e) {
+          console.log(e);
+        }
+      });
+
+      clipboardclient.on('copy', function(evt) {
+        var clipboard = evt.clipboardData;
+        clipboard.setData("text/plain", $(evt.target).text().trim());
+      });
+
+      clipboardclient.on('aftercopy', function(evt) {
+        var highlightedElement = evt.target;
+        $(highlightedElement).addClass('copied');
+        setTimeout(function(){
+          $(highlightedElement).removeClass('copied');
+        },1500);
+
+        _gaq.push(['_trackEvent', 'Emojis', 'Copy', text]);
+      });
+    });
 
-    clip.setHandCursor(true);
-    clip.addEventListener('complete', function(client, text) {
-      $.jnotify('Copied <code>' + text + '</code>');
-      _gaq.push(['_trackEvent', 'Emojis', 'Copy', text]);
+    clipboardclient.on( 'error', function(event) {
+      ZeroClipboard.destroy();
+      initJsClipboard();
     });
+  }  
 
-    $('ul').on('mouseover', 'div', function() {
+  var jsClipboardSupported = true; // we can't check if this is true unless the user tries once
+  function initJsClipboard() {
+    $('ul').on('click', 'div', function() {
       try {
-        if (clip_flash_block_detect.movie.PercentLoaded()) {
-          clip.setText($(this).text().trim());
-          if (clip.div && !reglue) {
-            clip.receiveEvent('mouseout', null);
-            clip.reposition(this);
+        if(jsClipboardSupported) {
```

Comment:
well, the first time there is an error, you report it with the ``clipboardError`` class. Maybe do the same for later trials ?